### PR TITLE
refact: reworks test framework integration 

### PR DESCRIPTION
**Pull request by pftg**
- Mock Capybara's approach to the setup project.
- Compressing the namespace path and dropping extra 2 nested modules
- Reduced dependency on the dynamic value constant
- Reduced minitest dependency if you are using RSpec
- Upgraded minitest with fixed version

### CODE REVIEW

**Comment by pftg** [comment]
@UweKubosch please check

**Review by donv — APPROVED** [review]

File: .github/actions/upload-screenshots/action.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+name: 'Setup Ruby and Cache'
+description: 'Set up Ruby environment and cache apt packages'
```
[the remark this answers is not in the document]
Comment (pftg) [reply .github/actions/upload-screenshots/action.yml:3]: Yeah, copy paste ;)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -67,15 +71,10 @@ jobs:
           SCREENSHOT_DRIVER: vips
         run: bin/rake test
 
-      - name: Upload Screenshots
-        if: always()
-        uses: actions/upload-artifact@v4
+      - name: Setup Ruby and Cache Apt Packages
```
[the remark this answers is not in the document]
Comment (pftg) [reply .github/workflows/test.yml:74]: yeah, nice notice

File: README.md [hunk 3]
```diff
@@ -40,44 +40,42 @@ Or install it yourself as:
 
 ### Minitest
 
-In your test class, include the `Capybara::Screenshot::Diff` module:
+In your test class, include the `CapybaraScreenshotDiff::Minitest::Assertions` module:
```
[the remark this answers is not in the document]
Comment (pftg) [reply README.md:43]: k

**Pull request merged**